src/include/mpir_datatype.h

```diff
@@ -132,6 +132,10 @@ struct MPIR_Datatype {
     /* pointer to contents and envelope data for the datatype */
     MPIR_Datatype_contents *contents;
 
+    /* flattened representation */
+    void *flattened;
+    int flattened_sz;
```

Comment:
Why does the `flattened_sz` need to be cached?

Reply:
I need to know how big it is to send it somewhere.

Reply:
Oh, sorry, yes.  I was just considering the unflattening part, which doesn't need it.